django/utils/functional.py

```diff
@@ -75,6 +75,10 @@ def __reduce__(self):
                 (func, self.__args, self.__kw) + resultclasses
             )
 
+        def __repr__(self):
+            """ Make sure the lazy object's display is useful when debugging """
+            return self.__cast()
```

Comment:
It think you should be returning `repr(self.__cast())` instead if the proxied object is not and instance of `str`.